cf/mixin/fielddomain.py

```diff
@@ -0,0 +1,2958 @@
+        cached = filter_kwargs.get("cached")
+        if cached is not None:
+            return cached
+
+        filter_kwargs["todict"] = True
+
+        c = getattr(self, _constructs_method)(*identities, **filter_kwargs)
+
+        # Return construct, or key, or both, or default
+        n = len(c)
+        if n == 1:
+            k, construct = c.popitem()
+            if key:
+                return k
+
+            if item:
+                return k, construct
+
+            return construct
+
+        if default is None:
+            return default
+
+        return self._default(
+            default,
+            f"{self.__class__.__name__}.{_method}() can't return {n} "
+            "constructs",
+        )
+
+    @_manage_log_level_via_verbosity
+    def _equivalent_coordinate_references(
+        self,
+        field1,
+        key0,
+        key1,
+        atol=None,
+        rtol=None,
+        s=None,
+        t=None,
+        verbose=None,
+        axis_map=None,
+    ):
+        """True if coordinate reference constructs are equivalent.
+
+        Two real numbers ``x`` and ``y`` are considered equal if
+        ``|x-y|<=atol+rtol|y|``, where ``atol`` (the tolerance on absolute
+        differences) and ``rtol`` (the tolerance on relative differences)
+        are positive, typically very small numbers. See the *atol* and
+        *rtol* parameters.
+
+        :Parameters:
+
+            ref0: `CoordinateReference`
+
+            ref1: `CoordinateReference`
+
+            field1: `Field`
+                The field which contains *ref1*.
+
+        :Returns:
+
+            `bool`
+
+        """
+        ref0 = self.coordinate_references(todict=True)[key0]
+        ref1 = field1.coordinate_references(todict=True)[key1]
+
+        if not ref0.equivalent(ref1, rtol=rtol, atol=atol, verbose=verbose):
+            logger.info(
+                f"{self.__class__.__name__}: Non-equivalent coordinate "
+                f"references ({ref0!r}, {ref1!r})"
+            )  # pragma: no cover
+            return False
+
+        # Compare the domain ancillaries
+        # TODO consider case of None key ?
+        for (
+            term,
+            identifier0,
+        ) in ref0.coordinate_conversion.domain_ancillaries().items():
+            if identifier0 is None:
+                continue
+
+            identifier1 = ref1.coordinate_conversion.domain_ancillaries()[term]
+
+            #            key0 = domain_ancillaries.filter_by_key(identifier0).key()
+            #            key1 = field1_domain_ancillaries.filter_by_key(identifier1).key()
+
+            if not self._equivalent_construct_data(
+                field1,
+                key0=identifier0,  # key0,
+                key1=identifier1,  # key1,
+                rtol=rtol,
+                atol=atol,
+                s=s,
+                t=t,
+                verbose=verbose,
+                axis_map=axis_map,
+            ):
+                # add traceback TODO
```

Comment:
There's a few TODOs for this function that might not have been addressed, in particular this one.

Reply:
Happy to leave this one till after the release, as the traceback is "merely" informative, rather than functional ...

Reply:
Sure, noted.